FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -0,0 +1,428 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Data.Tables;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Synapse.Common.Configurations;
+using Microsoft.Health.Fhir.Synapse.Common.Models.FhirSearch;
+using Microsoft.Health.Fhir.Synapse.Common.Models.Jobs;
+using Microsoft.Health.Fhir.Synapse.Core.DataFilter;
+using Microsoft.Health.Fhir.Synapse.Core.Exceptions;
+using Microsoft.Health.Fhir.Synapse.Core.Extensions;
+using Microsoft.Health.Fhir.Synapse.Core.Fhir;
+using Microsoft.Health.Fhir.Synapse.Core.Jobs.Models;
+using Microsoft.Health.Fhir.Synapse.Core.Jobs.Models.AzureStorage;
+using Microsoft.Health.Fhir.Synapse.DataClient;
+using Microsoft.Health.Fhir.Synapse.DataClient.Api;
+using Microsoft.Health.Fhir.Synapse.DataClient.Extensions;
+using Microsoft.Health.Fhir.Synapse.DataClient.Models.FhirApiOption;
+using Microsoft.Health.Fhir.Synapse.DataWriter;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using JobStatus = Microsoft.Health.JobManagement.JobStatus;
+
+namespace Microsoft.Health.Fhir.Synapse.Core.Jobs
+{
+    public class FhirToDataLakeOrchestratorJob : IJob
+    {
+        private readonly JobInfo _jobInfo;
+        private readonly FhirToDataLakeOrchestratorJobInputData _inputData;
+        private readonly IFhirDataClient _dataClient;
+        private readonly IFhirDataWriter _dataWriter;
```

Comment:
write to FHIR or ADLS?

Reply:
It writes to ADLS. the name here means the data source is FHIR, it writes `FhirData`. The implemented class named `AzureBlobDataWriter`, for other destinations, we can implement other `xxxDataWriter`

Reply:
Per our offline discussion, the design is OK.